common/src/main/scala/uk/ac/wellcome/platform/archive/common/ConvertibleToInputStream.scala

```diff
@@ -20,17 +21,21 @@ object ConvertibleToInputStream {
   implicit object ConvertibleToInputStreamObjectLocation
       extends ToInputStream[ObjectLocation] {
 
-    def apply(t: ObjectLocation)(implicit s3Client: AmazonS3,
-                                 ec: ExecutionContext): Future[InputStream] =
+    def apply(objectLocation: ObjectLocation)(
+      implicit s3Client: AmazonS3,
+      ec: ExecutionContext): Future[InputStream] =
       Future(
-        s3Client.getObject(t.namespace, t.key)
+        s3Client.getObject(objectLocation.namespace, objectLocation.key)
       ).map(
           response => response.getObjectContent
         )
         .recover {
           case throwable: Throwable =>
-            throw new RuntimeException(
-              s"Error getting input stream for s3://$t: ${throwable.getMessage}")
+            throw new InvalidObjectLocationException(
+              objectLocation = objectLocation,
+              message =
+                s"Error getting input stream for s3://$objectLocation: ${throwable.getMessage}",
```

Comment:
I think the more useful thing we could do here would be to distinguish between the most common S3 errors, and bubble up specific exceptions for those.

“No such object in S3” is one

“Do not have permissions to read object” is another

And this is where we need to make that distinction.

Reply:
👍 distinguishing these by status code.